os_migrate/plugins/module_utils/router_interface.py

```diff
@@ -58,30 +58,32 @@ def create_or_update(self, conn):
         # 1. ensure a port exists with the required `fixed_ips`
         sdk_params = self._to_sdk_params(refs)
         port = self._find_port_by_subnet_ips(conn, refs)
-
-        if port:
-            # port exists but is assigned to something else => fail
-            if ((port['device_owner'] and port['device_owner'] != 'network:router_interface')
-                    or (port['device_id'] and port['device_id'] != refs['device_id'])):
-                raise exc.CannotConverge(
-                    "Port with IP address for router import exists, and it is assigned "
-                    "to a different device: {0}".format(port.to_dict())
-                )
-
-            if self._port_needs_update(self.from_sdk(conn, port)):
-                port = self._update_port(conn, port['id'], sdk_params)
+        try:
+            if port:
+                # port exists but is assigned to something else => fail
+                if ((port['device_owner'] and port['device_owner'] != 'network:router_interface')
+                        or (port['device_id'] and port['device_id'] != refs['device_id'])):
+                    raise exc.CannotConverge(
+                        "Port with IP address for router import exists, and it is assigned "
+                        "to a different device: {0}".format(port.to_dict())
+                    )
+
+                if self._port_needs_update(self.from_sdk(conn, port)):
+                    port = self._update_port(conn, port['id'], sdk_params)
+                    changed = True
+            else:
+                port = self._create_port(conn, sdk_params)
                 changed = True
-        else:
-            port = self._create_port(conn, sdk_params)
-            changed = True
-
-        # 2. ensure the port is attached to the router by `router_name`
-        if port['device_owner'] != 'network:router_interface' \
-           and port['device_id'] != refs['device_id']:
-            conn.network.add_interface_to_router(
-                refs['device_id'], port_id=port['id'])
-            changed = True
 
+            # 2. ensure the port is attached to the router by `router_name`
+            if port['device_owner'] != 'network:router_interface' \
+               and port['device_id'] != refs['device_id']:
+                conn.network.add_interface_to_router(
+                    refs['device_id'], port_id=port['id'])
+                changed = True
+        except openstack.exceptions.ConflictException:
```

Comment:
There shouldn't be any conflict exception here:

* we're looking up the port by IPs, if it exists, we don't create it

* we're looking if it's assigned to the router already, if yes, we don't try to assign it again

The method should be a no-op, something's not working. I think we should figure out what's being executed that shouldn't be executed, rather than catching the exception.

Reply:
The error message output is here: https://gist.github.com/rbrady/ff2b8f41cb3a6738599102478d900078